### PR TITLE
Add font settings

### PR DESCRIPTION
**Pull request by radiden**
first time working with vue, hopefully didn't commit too many crimes against code quality
image

### CODE REVIEW

File: src/editor/editor.js [hunk 1]
```diff
@@ -121,6 +121,14 @@ export class HeynoteEditor {
         })
     }
 
+    setFontFamily(fontFamily) {
+        fontFamily ? this.view.dom.style.fontFamily = fontFamily : this.view.dom.style.fontFamily = "Hack, Menlo, Monaco, 'Courier New', monospace"
+    }
+
+    setFontSize(fontSize) {
+        fontSize ? this.view.dom.style.fontSize = fontSize : this.view.dom.style.fontSize = "12px"
```
Comment (felipelube) [inline src/editor/editor.js:129]:
```suggestion
        this.view.dom.style.fontSize = fontSize || "12px"
```

File: src/editor/editor.js [hunk 2]
```diff
@@ -121,6 +121,14 @@ export class HeynoteEditor {
         })
     }
 
+    setFontFamily(fontFamily) {
+        fontFamily ? this.view.dom.style.fontFamily = fontFamily : this.view.dom.style.fontFamily = "Hack, Menlo, Monaco, 'Courier New', monospace"
```
Comment (felipelube) [inline src/editor/editor.js:125]:
```suggestion
        this.view.dom.style.fontFamily = fontFamily || "Hack, Menlo, Monaco, 'Courier New', monospace"
```

**Comment by heyman** [comment]
Hi! At the moment, I'm hesitant about adding font settings to Heynote. It can cause UI issues that is hard to catch, and it adds maintenance burden to make sure everything looks good with custom fonts.

**Comment by keerthik** [comment]
Just throwing my +1 for having basic font settings. Even just a few step sizes that are easier to QA without being able to change in font family would be fine, like "small"/"medium"/"large". I use my 4K monitors at a decent distance and the text is 20% smaller than what is comfortable for me to read, especially for something I would like to skim quickly.

**Comment by heyman** [comment]
> Even just a few step sizes that are easier to QA without being able to change in font family would be fine, like "small"/"medium"/"large". I use my 4K monitors at a decent distance and the text is 20% smaller than what is comfortable for me to read, especially for something I would like to skim quickly.

Definitely more inclined to add a setting like that. Is there any reason for why you don't want to use the zoom feature (assuming the zoom-level was saved between restarts)?

**Comment by dhruvkb** [comment]
@heyman while I appreciate your focus in keeping the app minimal and also your concern about how fonts can make UI issues hard to catch, I believe font customization is a very important feature. As developers, we spend hours trying to find our perfect font that has the look, readability and other features we need (like ligatures, powerline icons or custom glyphs).

Font choice is deeply personal and individual. Personally I would love to use the code font I use in my code editor for Hey Note because I've tweaked it to be perfect for that purpose. But currently I cannot do so.

As for the added maintenance burden, we'd welcome the font customisation on the precondition that issues arising from custom fonts are out of scope, low priority or even wontfix.

**Comment by keerthik** [comment]
> > Even just a few step sizes that are easier to QA without being able to change in font family would be fine, like "small"/"medium"/"large". I use my 4K monitors at a decent distance and the text is 20% smaller than what is comfortable for me to read, especially for something I would like to skim quickly.
> 
> Definitely more inclined to add a setting like that. Is there any reason for why you don't want to use the zoom feature (assuming the zoom-level was saved between restarts)?
> 
> Edit: The zoom-level _is_ saved between restarts.

This would work for me today if...it worked. I am looking at the windows version of Heynote Version 1.4.2 (beta channel) right now. The Window > Zoom button doesn't do anything I can see. There's no response to standard shortcuts either, like (Ctrl + =) for zoom in or (Ctrl + -) to zoom out, or (Ctrl + mouse-scroll-up/down).

Is this a known issue or am I confused about where to get Zoom from?

**Comment by heyman** [comment]
I've changed my mind on font settings and implemented it in #147.

**Pull request closed without merging**